arcgis-ios-sdk-samples/Maps/Display device location with NMEA data sources/DisplayDeviceLocationWithNMEADataSourcesViewController.swift

```diff
@@ -26,29 +26,90 @@ class DisplayDeviceLocationWithNMEADataSourcesViewController: UIViewController {
     }
     /// The label to display satellites info.
     @IBOutlet var statusLabel: UILabel!
-    /// The button to start the demo.
-    @IBOutlet var startBarButtonItem: UIBarButtonItem!
+    /// The button to choose a data source and start the demo.
+    @IBOutlet var sourceBarButtonItem: UIBarButtonItem!
     /// The button to reset pan mode to "recenter".
     @IBOutlet var recenterBarButtonItem: UIBarButtonItem!
     /// The button to reset the demo.
     @IBOutlet var resetBarButtonItem: UIBarButtonItem!
     
+    // MARK: Constants
+    
+    /// The protocols specified in the `Info.plist` that the app uses to
+    /// communicate with external accessory hardware.
+    let supportedProtocolStrings: [String] = {
+        guard let protocols = Bundle.main.object(forInfoDictionaryKey: "UISupportedExternalAccessoryProtocols") as? [String] else {
+            return []
+        }
+        return protocols
+    }()
+    
     // MARK: Instance properties
     
     /// An NMEA location data source, to parse NMEA data.
-    let nmeaLocationDataSource = AGSNMEALocationDataSource(receiverSpatialReference: .wgs84())
+    var nmeaLocationDataSource: AGSNMEALocationDataSource?
+    /// A string to hold the latest satellite info.
+    var satelliteInfoText = "Satellites info will be shown here."
```

Comment:
In general it is best not to pre-compute strings displayed to the user, particularly if the string is locale-dependent (ie. was created from a formatter) as information in the locale may have changed before the string is actually displayed onscreen. For example, a user crossing into a different time zone or region. That may not be an issue if the string is displayed right away, but then why have a property for it? Another thing to consider is that when a label's text is updated, the entire label needs to be redrawn, even if only a small part of that text changed. Instead of trying to put multiple pieces of information in a single label, it would be better to have a separate label for each piece. That way, when on one piece of information changes (like the horizontal accuracy, for example), only that label needs to be re-drawn. Does that make sense?

Reply:
I slacked off here 😅  will use 2 labels for different text!

Reply:
> Another thing to consider is that when a label's text is updated, the entire label needs to be redrawn, even if only a small part of that text changed. Instead of trying to put multiple pieces of information in a single label, it would be better to have a separate label for each piece.

In terms of this issue - I totally agree it is the right approach, but am a bit reluctant to do so. It may improve the performance of the sample, but would make it harder to read the source code as the users need to find out the mappings between labels and static/mutating text pieces. In contrast, putting all the prompt text in a single string and only assign to one label is clearer IMO.

It's pushing me to think of Swift UI's `Text`... 🤐

Reply:
> It may improve the performance of the sample, but would make it harder to read the source code

The suggestion ("Instead of trying to put multiple pieces of information in a single label, it would be better to have a separate label for each piece.") implies that each piece of information is updated independently. When all pieces are updated at the same time, then yes, it does make sense to put them in the same label. I think what you've done with having a separate label for the accuracy is a good improvement and putting everything else in the other label makes sense.